sdk/template/Azure.Template/tests/MiniSecretClientLiveTests.cs

```diff
@@ -30,7 +38,7 @@ public async Task CanGetSecret()
 
             var secret = await client.GetSecretAsync("TestSecret");
 
-            Assert.AreEqual("Very secret value", secret.Value.Value);
+            Assert.AreEqual(TestEnvironment.KeyVaultSecret, secret.Value.Value);
```

Comment:
Would need to re-record to add the `KEYVAULT_SECRET` to the recording session, but I agree with @pakrym offline that we probably shouldn't be asserting the secret value anyway. Instead, maybe just change this to `Assert.NotNull` just to show an example of how we are asserting the response in some way.

Reply:
Is the concern that it is generally bad practice to expose a secret in tests, or something else? If the tradeoff is between appearing more secure and making the assert more useful, I'd vote the latter.

Reply:
In playback, you end up comparing "Sanitized" to "Sanitized"

Reply:
Why would it be sanitized? It's not really a secret.

Reply:
Which would be fine. The main reason I did this was to show an example in the TextEnvironment how to mark something a secret. I was talking with @pakrym offline about this and suggested a custom sanitizer with a JSONPath to sanitize secrets. The more I think about it, the more I think it's a good idea to add - with appropriate comments this is only an example you can do for secrets (in the general sense) but not always necessary.

My goal was to add common patterns in the template to serve as an example. Doing that, we can have very specific samples in code, and just update the guidelines to be vaguer with links to the code if necessary.

Reply:
I think a realistic customer sanitizer makes a lot of sense also. Maybe one of each type (header, body json path, url, etc)

Reply:
Perhaps. I didn't think we needed to cover every possibility - that's more to delete from the template if not needed - but just making the sanitizer customization known can lead them down a road of discovery.

Reply:
One thing I would say is don't overload the template with optional stuff. It's hard for a person seeing SDK tests for the first time to know what's required and what's not. There is a high probability that they'll just keep random unused things in.

We should make sure that information about features is easy to find and understand when needed but sticking every possible feature into the template is not the way to go, IMO.

Reply:
That makes sense - Perhaps the template base test class can just include a few high level comments pointing to key concepts like this in our readme.

Reply:
Sure. I could effectively change that attribute to a comment, but I imagine that - in practice - most client libraries won't support only one version for long. I included this originally for the same reason as some of the other stuff in this PR: advertisement. But a comment could serve just fine. My only con I'll mentoin, though, is that the comment will also have to tell them to add the ctor parameter and actually use it in their `CreateClient` method. That's why I'm more leaning toward keeping this. It doesn't hurt anything.

Reply:
Discussed some changes offline with @pakrym. Will mostly link to the TestFramework README (didn't know it existed).